### ISSUE TITLE
NFTs/OpenSea

### ISSUE THREAD

**Issue by Karotoka** [LeXpunK-Army/infra-bill#3]
I think we need to turn our mind to whether or not NFTs fall into this definition. Ideally, they do not. But my read of this 

Also note that the proposed definition of "Major Digital Asset Exchange" would include NFT hubs like OpenSea and Rarible so long as NFTs are "Digital Assets". I don't know if that's a positive outcome as it may put securities requirements on a 'broker' who are poorly equipped to deal with AML.

I propose adding 'fungible' to the definition of Digital Asset to try and carve our a separate space for NFTs.

**Comment by gonbegood** [comment]
Just to test the underlying point of principle - why should an NFT be treated differently to another digital asset (from a legal property perspective)? I agree that there may be different treatment from a practical/regulatory perspective.

**Comment by larryflorio** [comment]
@MLKimber because fungibility is fundamental to both commodities and securities markets, and it's why we're able to have liquid markets for tokens. One DAI is the same as any other DAI. 

To take it out of the crypto context, including NFTs in this regulation would like saying trading cards are securities and a business selling them should be regulated as an exchange.

**Comment by gonbegood** [comment]
I agree with the point in the context of the wider purpose of the bill and wasn't suggesting that NFTs should be included within its scope. I was just suggesting that the term Digital Asset could be defined more broadly, and then the specific definition / sub-definition put forward here could be used to define those particular Digital Assets that the bill is aimed at.

**Comment by lex-node** [comment]
the problem is that "NFT" is not an asset type, it's a data structure type

a stock certificate can be represented by an NFT, or even a governance token can be issued in the form of NFTs. . .so I think carving out all "NFTs" would probably be too broad of a carve-out--would open an easy loophole and everyone would just start issuing all their valuable tokens as NFTs

What you really mean to carve-out is NFT-based art collectibles. Which I think does make sense to carve out, but OTOH kind of opens up a pandora's box (what other things should we carve out? becomes a laundry list and gets into the same issues we saw with CoincCenter's proposed carve-outs). . .

**Comment by gonbegood** [comment]
I agree with you that an NFT (regardless of what it does/is "tethered"/"stapled" to) is just a data structure type. And I agree that is quite hard to distinguish from other tokens/"cryptocurrencies", in a definition without getting too specific. And as you say, people would just structure their assets as NFTs to avoid the regulation if they're not included. 

Which is why I like your approach of starting wide, based on defining the data structure type and then carving-in what you actually want to regulate/talk about. The data structure type is novel because it operates to mimic what existing property law does/what existing contractual arrangements in intermediated securities do. 

If you did want to start with a broader definition, you could use something like the following, which is based on your work in SCoDA (which I really liked). That work linked the record of information to its function within the protocol (i.e. (1) the ability to associate the data string with a location in geometric space - an address, and (2) the ability to use/transfer). Incidentally, that is how property law operates - there is (i) an object (here a data structure/data object); (ii) which a person has a liberty to use (here, to have their address associated with the object and transact); and (iii) which the law protects by a legal right to exclude others (here that legal right is mimicked in a practical sense in the ability to exclude others by keeping your private key secret/safe). 

I've also used the term Digital Object to separate any "value" or "asset" considerations in the broad definition. And because from a technical perspective I think the instantiated data - the data structure with specific functionality - could be described as a data object (but you'll correct me on that if I'm wrong).  

The below definition at least gives you the flexibility to tweak the definitions of Protocol Rules and Digital Object Network to be more inclusive or exclusive, as you like. You could also add a definition of “Control” which would align with the work the UCC are doing re: “Controllable Electronic Records”. 

You could then go on to limit back down the scope of the _operative_ provisions of the infrastructure bill itself in the way you already do, e.g. by reference to specific Digital Objects that function in certain ways. So you could carve-in representations of legal tender/securities, and carve-in Digital Objects which are “tethered” or “stapled” to things, or rights in relating to legal tender/securities. You can then also carve-in things based on practical, people based outcomes (like you do with trade volume) rather than simply including a Digital Object within the scope just because of how the particular data string is constructed.

The below is pretty rough and is based on your work so I didn't include as a pull-request because you might be thinking about things differently/prefer your SCoDA version. 

**Digital Object** means any digital data string, record or unit of account which is recognised by the Protocol Rules of a Digital Object Network as capable of: 

1. being uniquely associated with (or “owned”) by a particular Digital Address at any one time (including by reference to the relevant timechain or block height, if applicable); and
2. being uniquely associated with (or “controlled”) by a particular digital string (including a private key or group of private keys) at any one time, such that the associated digital data string (or a data string derived from it) is required to effect any state change (including any Transfer) to the Digital Object Network.

**Confirmed** means a change of state of the distributed digital ledger or digital data structure in accordance with the Protocol Rules of the Digital Object Network which has become probabilistically irreversible.

**Digital Address** means a particular digital data string (including a public-key address) which is recognised by the relevant Digital Object Network.

**Digital Object Network** means a distributed digital ledger or digital data structure in which consensus as to the state of the distributed digital ledger or digital data structure is achieved in accordance with the Protocol Rules of that Digital Object Network.  

**Protocol Rules** means a mathematically verifiable and fault-tolerant process which is followed by participating nodes in the Digital Object Network.

A **Transfer** of a Digital Object to a given Digital Address will be deemed to have occurred if: 

1. a new transfer function has been submitted to the Digital Object Network that:

(A) provides for the re-association of Digital Object with the receiving Digital Address (such “re-association” to include any replacement, modification, destruction, cancellation, or elimination of a Digital Object which results in a corresponding derivative creation and acquisition of a new Digital Object or a or derivative of that Digital Object); and
 (B) is signed by a particular digital or electronic data string (including a private key or group of private keys) which is recognised by the relevant Digital Object Network as sufficient to authorize the execution of such transfer function; and 

2. such transaction is Confirmed.

**Comment by Karotoka** [comment]
> the problem is that "NFT" is not an asset type, it's a data structure type
> 
> a stock certificate can be represented by an NFT, or even a governance token can be issued in the form of NFTs. . .so I think carving out all "NFTs" would probably be too broad of a carve-out--would open an easy loophole and everyone would just start issuing all their valuable tokens as NFTs
> 
> What you really mean to carve-out is NFT-based art collectibles. Which I think does make sense to carve out, but OTOH kind of opens up a pandora's box (what other things should we carve out? becomes a laundry list and gets into the same issues we saw with CoincCenter's proposed carve-outs). . .

My concern is less about the NFTs themselves and more about the OpenSeas of the community. I'm going to have to pay cap gains on my penguin, just like I would if it was a token (again, I'm Canadian so may be different in US). The big difference is the requirements of the broker or facilitator of the art houses. Does OpenSea require AML/KYC? Should it? I think by going with your original definition we cede that ground without a fight.

I also think you are a little too optimistic on the loophole such 'fungibility' term would open. Yes, you could NFT a share cert, but I think the actual interpretation of that is that you have made an NFT or representation of a basket of rights that actually reside with the holder of the underlying security. The NFT of the share cert may be that representation or it may not but that's a fact determination.

**Comment by Karotoka** [comment]
> I agree with you that an NFT (regardless of what it does/is "tethered"/"stapled" to) is just a data structure type. And I agree that is quite hard to distinguish from other tokens/"cryptocurrencies", in a definition without getting too specific. And as you say, people would just structure their assets as NFTs to avoid the regulation if they're not included.
> 
> Which is why I like your approach of starting wide, based on defining the data structure type and then carving-in what you actually want to regulate/talk about. The data structure type is novel because it operates to mimic what existing property law does/what existing contractual arrangements in intermediated securities do.
> 
> If you did want to start with a broader definition, you could use something like the following, which is based on your work in SCoDA (which I really liked). That work linked the record of information to its function within the protocol (i.e. (1) the ability to associate the data string with a location in geometric space - an address, and (2) the ability to use/transfer). Incidentally, that is how property law operates - there is (i) an object (here a data structure/data object); (ii) which a person has a liberty to use (here, to have their address associated with the object and transact); and (iii) which the law protects by a legal right to exclude others (here that legal right is mimicked in a practical sense in the ability to exclude others by keeping your private key secret/safe).
> 
> I've also used the term Digital Object to separate any "value" or "asset" considerations in the broad definition. And because from a technical perspective I think the instantiated data - the data structure with specific functionality - could be described as a data object (but you'll correct me on that if I'm wrong).
> 
> The below definition at least gives you the flexibility to tweak the definitions of Protocol Rules and Digital Object Network to be more inclusive or exclusive, as you like. You could also add a definition of “Control” which would align with the work the UCC are doing re: “Controllable Electronic Records”.
> 
> You could then go on to limit back down the scope of the _operative_ provisions of the infrastructure bill itself in the way you already do, e.g. by reference to specific Digital Objects that function in certain ways. So you could carve-in representations of legal tender/securities, and carve-in Digital Objects which are “tethered” or “stapled” to things, or rights in relating to legal tender/securities. You can then also carve-in things based on practical, people based outcomes (like you do with trade volume) rather than simply including a Digital Object within the scope just because of how the particular data string is constructed.
> 
> The below is pretty rough and is based on your work so I didn't include as a pull-request because you might be thinking about things differently/prefer your SCoDA version.
> 
> **Digital Object** means any digital data string, record or unit of account which is recognised by the Protocol Rules of a Digital Object Network as capable of:
> 
> 1. being uniquely associated with (or “owned”) by a particular Digital Address at any one time (including by reference to the relevant timechain or block height, if applicable); and
> 2. being uniquely associated with (or “controlled”) by a particular digital string (including a private key or group of private keys) at any one time, such that the associated digital data string (or a data string derived from it) is required to effect any state change (including any Transfer) to the Digital Object Network.
> 
> **Confirmed** means a change of state of the distributed digital ledger or digital data structure in accordance with the Protocol Rules of the Digital Object Network which has become probabilistically irreversible.
> 
> **Digital Address** means a particular digital data string (including a public-key address) which is recognised by the relevant Digital Object Network.
> 
> **Digital Object Network** means a distributed digital ledger or digital data structure in which consensus as to the state of the distributed digital ledger or digital data structure is achieved in accordance with the Protocol Rules of that Digital Object Network.
> 
> **Protocol Rules** means a mathematically verifiable and fault-tolerant process which is followed by participating nodes in the Digital Object Network.
> 
> A **Transfer** of a Digital Object to a given Digital Address will be deemed to have occurred if:
> 
> 1. a new transfer function has been submitted to the Digital Object Network that:
> 
> (A) provides for the re-association of Digital Object with the receiving Digital Address (such “re-association” to include any replacement, modification, destruction, cancellation, or elimination of a Digital Object which results in a corresponding derivative creation and acquisition of a new Digital Object or a or derivative of that Digital Object); and
> (B) is signed by a particular digital or electronic data string (including a private key or group of private keys) which is recognised by the relevant Digital Object Network as sufficient to authorize the execution of such transfer function; and
> 
> 1. such transaction is Confirmed.

This is mega. 

I really appreciate the idea of trying to set out a protocol definition so that you can build things off of it (e.g. transfer, confirmation). I would actually prefer this approach (starting narrow and broadening) rather than the starting wide and narrowing. I think you'll have major issues trying to have a politician/staff read that and understand what the hell it is saying but that is why we need to fund advocacy and education.

> **Digital Object Network** means a distributed digital ledger or digital data structure in which consensus as to the state of the distributed digital ledger or digital data structure is achieved in accordance with the Protocol Rules of that Digital Object Network.

Maybe for this we cut everything after "...structure is achieved" as the Protocol Rules should be implicit in how that confirmation takes place in the first place. Perhaps change "achieved" to "confirmed"?

I really do like the idea of trying to make crypto regulations a 'new' thing and defining it as such. It is much better than the "this is a typical security except for all these dozens of differences".

**Comment by lex-node** [comment]
@MLKimber , I like this [this](https://github.com/LeXpunK-Army/infra-bill/issues/3#issuecomment-898598671) a lot. Similar to what I did with SCODA but more implementation-neutral/universal. 

You should consider building this out into a set of legal semantics primitives for talking about this technology. I suggest setting up a separate repo so people can comment / iterate more on your great starting point.

**Comment by gonbegood** [comment]
@lex-node thanks - have created a new repo [here](https://github.com/gonbegood/Crypto---Legal-Semantic-Primes).
